### PR TITLE
Add multiple path support

### PR DESCRIPTION
**Pull request by sobakasu**
Additional paths can be password protected by adding configuration to an initializer. e.g.

    Katalyst::Basic::Auth.add("/secure-path", username: "user", password: "pass") if Rails.env.production?

paths can also be excluded from basic auth, e.g.

    Katalyst::Basic::Auth.exclude("/public-path")

### CODE REVIEW

File: spec/support/rails.rb [hunk 1]
```diff
@@ -11,3 +11,31 @@ class DummyRailsEnv
     end
   end
 end
+
+class DummyRails
+  module VERSION
+    MAJOR = 4
+  end
```
Comment (SughoshRao) [inline spec/support/rails.rb:18]: Can we have a dummy app for testing Rails 6 as well?

**Review by SughoshRao — APPROVED** [review]
This is LEGTM

File: lib/katalyst/basic/auth/config.rb [hunk 2]
```diff
@@ -6,21 +6,61 @@ module Katalyst
...
+               .detect { |i| path.match(/^#{i.path}/) } || global
+          end
+
+          # @return [Config] The global configuration
+          def global
+            all[0]
+          end
+
+          def add(path:, username: nil, password: nil, enabled: nil)
+            config = Config.new(path: path, username: username, password: password, enabled: enabled)
```
Comment (mattr) [inline lib/katalyst/basic/auth/config.rb]: (opt) we're at the class level, so you shouldn't need to specify the class:
```
config = new(...)
```

**Review by mattr — APPROVED** [review]
LGTM

File: lib/katalyst/basic/auth/config.rb (hunk 2 above)
Comment (sobakasu) [reply lib/katalyst/basic/auth/config.rb]: good pick up, fixed

File: spec/support/rails.rb (hunk 1 above)
Comment (sobakasu) [reply spec/support/rails.rb:18]: not sure how I feel about dummy apps

Comment (sobakasu) [reply spec/support/rails.rb:18]: do people like dummy apps?

**Pull request merged**